src/Microsoft.ML.TensorFlow/TensorTypeExtensions.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using NumSharp.Backends;
+using NumSharp.Backends.Unmanaged;
+using NumSharp.Utilities;
+using Tensorflow;
+
+namespace Microsoft.ML.Transforms
+{
+    [BestFriend]
+    internal static class TensorTypeExtensions
+    {
+        public static void ToArray<T>(this Tensor tensor, T[] array) where T: unmanaged
+        {
+            var size = array.Length;
+            if (typeof(T).as_dtype() == tensor.dtype)
+            {
+                if (tensor.NDims == 0 && size == 1)
+                {
+                    unsafe
+                    {
+                        array[0] = *(T*)tensor.buffer;
+                    }
+                }
+
+                unsafe
+                {
+                    var len = (long)size;
+                    fixed (T* dst = array)
+                    {
+                        var src = (T*)tensor.buffer;
+                        len *= ((long)tensor.itemsize);
+                        System.Buffer.MemoryCopy(src, dst, len, len);
+                    }
+                }
+            }
+            else
+            {
+                if (tensor.NDims == 0 && size == 1) //is it a scalar?
+                {
+                    unsafe
+                    {
+#if _REGEN
+                        #region Compute
+		                switch (dtype.as_numpy_dtype().GetTypeCode())
+		                {
+			                %foreach supported_dtypes,supported_dtypes_lowercase%
```

Comment:
These variables are stored [here](https://github.com/SciSharp/NumSharp/blob/master/src/NumSharp.Core/globals.regen)

I would copy them locally, the template will look like this:
```C#
%supported_dtypes = ["Boolean","Byte","Int16","UInt16","Int32","UInt32","Int64","UInt64","Char","Double","Single","Decimal"]
%supported_dtypes_lowercase = ["bool","byte","short","ushort","int","uint","long","ulong","char","double","float","decimal"]
%foreach supported_dtypes,supported_dtypes_lowercase%
```